backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java

```diff
@@ -126,33 +147,47 @@ public Endpoint getOrCreateEmailSubscriptionEndpoint(String accountId, EmailSubs
         return createEndpoint(endpoint);
     }
 
-    public Long getEndpointsCountPerCompositeType(String tenant, @Nullable String name, Set<CompositeEndpointType> type, Boolean activeOnly) {
-        return EndpointRepository.queryBuilderEndpointsPerType(tenant, name, type, activeOnly)
+    public Long getEndpointsCountPerCompositeType(String tenant, String orgId, @Nullable String name, Set<CompositeEndpointType> type, Boolean activeOnly) {
+        return EndpointRepository.queryBuilderEndpointsPerType(tenant, orgId, featureFlipper.isUseOrgId(), name, type, activeOnly)
                 .buildCount(entityManager::createQuery)
                 .getSingleResult();
     }
 
-    public QueryBuilder<Endpoint> getEndpointsQuery(String tenant, @Nullable String name) {
-        return QueryBuilder.builder(Endpoint.class)
-                .alias("e")
-                .where(
-                        WhereBuilder.builder()
-                                .and("e.accountId = :accountId", "accountId", tenant)
-                                .ifAnd(
-                                        name != null && !name.isEmpty(),
-                                        "LOWER(e.name) LIKE :name",
-                                        "name", (Supplier<String>) () -> "%" + name.toLowerCase() + "%"
-                                )
-                );
+    public QueryBuilder<Endpoint> getEndpointsQuery(String tenant, String orgId, @Nullable String name) {
+        if (orgIdHelper.useOrgId(orgId)) {
+            return QueryBuilder.builder(Endpoint.class)
+                    .alias("e")
+                    .where(
+                            WhereBuilder.builder()
+                                    .and("e.orgId = :orgId", "orgId", orgId)
+                                    .ifAnd(
+                                            name != null && !name.isEmpty(),
+                                            "LOWER(e.name) LIKE :name",
+                                            "name", (Supplier<String>) () -> "%" + name.toLowerCase() + "%"
+                                    )
+                    );
+        } else {
+            return QueryBuilder.builder(Endpoint.class)
+                    .alias("e")
+                    .where(
+                            WhereBuilder.builder()
+                                    .and("e.accountId = :accountId", "accountId", tenant)
```

Comment:
This will go away hopefully soon, but I would have used:
```
.ifAnd(orgIdHelper.useOrgId(orgId), "e.orgId = :orgId", "orgId", orgId)
.ifAnd(!orgIdHelper.useOrgId(orgId), e.accountId = :accountId", "accountId", tenant)
```

Reply:
Yeah I hesitated but then I realized the current (and ugly) code is easier to clean after the migration.